modules/data-streams/src/main/java/org/elasticsearch/datastreams/action/GetDataStreamsTransportAction.java

```diff
@@ -213,6 +203,28 @@ static GetDataStreamAction.Response innerOperation(
         );
     }
 
+    private static void collectIndexSettingsValues(
+        DataStream dataStream,
+        Map<Index, IndexProperties> backingIndicesSettingsValues,
+        Metadata metadata,
+        List<Index> backingIndices
+    ) {
+        for (Index index : backingIndices) {
+            IndexMetadata indexMetadata = metadata.index(index);
+            Boolean preferIlm = PREFER_ILM_SETTING.get(indexMetadata.getSettings());
+            assert preferIlm != null : "must use the default prefer ilm setting value, if nothing else";
+            ManagedBy managedBy;
+            if (metadata.isIndexManagedByILM(indexMetadata)) {
+                managedBy = ManagedBy.ILM;
+            } else if (dataStream.isIndexManagedByDataStreamLifecycle(index, metadata::index)) {
```

Comment:
This method (`DataStream::isIndexManagedByDataStreamLifecycle`) always returns false for failure indices because it immediately returns false if the index given is not one of the backing (non-failure) indices for the data stream.

Reply:
You can disregard this comment. This is planned to be in a later PR. It's actually correct to report that the failure indices are unmanaged because they are currently ignored by DataStreamLifecycleService.